contracts/integrations/test/internal-integration-tests/match_orders_test.ts

```diff
@@ -0,0 +1,2754 @@
+                takerAssetAmount: new BigNumber(98),
+            });
+            const signedOrderRight = await makerRight.signOrderAsync({
+                makerAssetAmount: new BigNumber(75),
+                takerAssetAmount: new BigNumber(13),
+            });
+
+            // Assert is rounding error ceil & not rounding error floor
+            // These assertions are taken from MixinMatchOrders::calculateMatchedFillResults
+            // The rounding error is derived by computing how much the left maker will sell.
+            const numerator = signedOrderLeft.makerAssetAmount;
+            const denominator = signedOrderLeft.takerAssetAmount;
+            const target = signedOrderRight.makerAssetAmount;
+            const _isRoundingErrorCeil = isRoundingErrorCeil(numerator, denominator, target);
+            expect(_isRoundingErrorCeil).to.be.true();
+            const _isRoundingErrorFloor = isRoundingErrorFloor(numerator, denominator, target);
+            expect(_isRoundingErrorFloor).to.be.false();
+
+            // Match signedOrderLeft with signedOrderRight
+            // Note that the left maker received a slightly better sell price.
+            // This is intentional; see note in MixinMatchOrders.calculateMatchedFillResults.
+            // Because the left maker received a slightly more favorable sell price, the fee
+            // paid by the left taker is slightly higher than that paid by the left maker.
+            // Fees can be thought of as a tax paid by the seller, derived from the sale price.
+            const expectedTransferAmounts = {
+                // Left Maker
+                leftMakerAssetSoldByLeftMakerAmount: new BigNumber(13),
+                leftMakerFeeAssetPaidByLeftMakerAmount: toBaseUnitAmount('76.4705882352941176', 16), // 76.47%
+                // Right Maker
+                rightMakerAssetSoldByRightMakerAmount: new BigNumber(75),
+                rightMakerFeeAssetPaidByRightMakerAmount: toBaseUnitAmount(100, 16), // 100%
+                // Taker
+                leftTakerFeeAssetPaidByTakerAmount: toBaseUnitAmount('76.5306122448979591', 16), // 76.53%
+                rightTakerFeeAssetPaidByTakerAmount: toBaseUnitAmount(100, 16), // 100%
+                leftProtocolFeePaidByTakerInEthAmount: DeploymentManager.protocolFee,
+                rightProtocolFeePaidByTakerInEthAmount: DeploymentManager.protocolFee,
+            };
+
+            await matchOrderTester.matchOrdersAndAssertEffectsAsync(
+                {
+                    leftOrder: signedOrderLeft,
+                    rightOrder: signedOrderRight,
+                },
+                expectedTransferAmounts,
+                matcher.address,
+                DeploymentManager.protocolFee.times(2),
+                false,
+            );
+        });
+
+        it('should transfer correct amounts when left order is fully filled and values pass isRoundingErrorCeil but fail isRoundingErrorFloor', async () => {
```

Comment:
These rounding tests (here and below) should probably be moved into unit tests. Optional nit to add a todo note.

Reply:
Yeah, that would make sense. Probably out of scope for this PR, but I'll add a note.